src/plugins/kibana/server/lib/ingest/processors/convert/__tests__/converter.js

```diff
@@ -0,0 +1,158 @@
+import expect from 'expect.js';
+import _ from 'lodash';
+import { kibanaToEs, esToKibana } from '../converter';
+
+describe('ingest', () => {
+
+  describe('processors', () => {
+
+    describe('converters', () => {
+
+      describe('convert', () => {
+
+        describe('kibanaToEs', () => {
+
+          let source;
+          let expected;
+          beforeEach(function () {
+            source = {
+              typeId: 'convert',
+              processor_id: 'foo_tag',
+              source_field: 'foo_field',
+              target_field: 'foo_target_field',
+              type: 'auto'
+            };
+
+            expected = {
+              convert: {
+                tag: 'foo_tag',
+                field: 'foo_field',
+                target_field: 'foo_target_field',
+                type: 'auto'
+              }
+            };
+          });
+
+          it('should convert from a kibana api object to an elasticsearch object', () => {
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
+
+          it('should ignore additional source fields', () => {
+            source.foo = 'bar';
+            source.bar = 'baz';
+
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
+
+          it('should map [auto] type correctly', () => {
+            source.type = 'auto';
+            expected.convert.type = 'auto';
+
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
+
+          it('should map [number] type correctly', () => {
+            source.type = 'number';
+            expected.convert.type = 'float';
+
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
+
+          it('should map [string] type correctly', () => {
+            source.type = 'string';
+            expected.convert.type = 'string';
+
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
+
+          it('should map [boolean] type correctly', () => {
+            source.type = 'boolean';
+            expected.convert.type = 'boolean';
+
+            const actual = kibanaToEs(source);
+            expect(_.isEqual(actual, expected)).to.be.ok();
+          });
```

Comment:
I'm not sure we need to test all the basic cases. IMO it draws the reader's attention away from the special `number` case so that they might miss it, and that's the really important one. If you wanna keep the basic cases, what would you say to putting them all in a single test case with multiple assertions? So then there would be two cases, one called `should map [number] to float` and another called `should leave basic types unmodified` or something like that.